src/main/java/org/reactome/release/downloaddirectory/MapOldStableIds.java

```diff
@@ -79,50 +83,107 @@ public static void execute(String username, String password, String host, String
 			{
 				deferredIds.add(stableIdsArray);
 			} else {
-				preferredIds.add(stableIdsArray);
+				stableIdsToOldIdsMappings.add(stableIdsArray);
 			}
 		}
-		preferredIds.addAll(deferredIds);
+		stableIdsToOldIdsMappings.addAll(deferredIds);
+
+		logger.info("Retrieving current stable identifiers from " + dba.getDBName());
+		Set<String> currentStableIdentifiers = getCurrentStableIdentifiers(dba);
 
 		// Write to file
-		File oldIdMappingFile = new File(filename);
-		if (oldIdMappingFile.exists()) {
-			oldIdMappingFile.delete();
-		}
-		oldIdMappingFile.createNewFile();
-		String header = "# Reactome stable IDs for release " + releaseNumber + "\n" + "Stable_ID\told_identifier(s)\n";
-		Files.write(Paths.get(filename), header.getBytes(), StandardOpenOption.APPEND);
-		for (List<Object> stableIdsArray : preferredIds)
-		{
-			String primaryId = (String) stableIdsArray.get(0);
-			@SuppressWarnings("unchecked")
-			List<String> secondaryIds = (ArrayList<String>) stableIdsArray.get(1);
-			String line = primaryId + "\t" + String.join(",", secondaryIds) + "\n";
-			Files.write(Paths.get(filename), line.getBytes(), StandardOpenOption.APPEND);
-		}
-		String outpathName = releaseNumber + "/" + filename;
-		Files.move(Paths.get(filename), Paths.get(outpathName), StandardCopyOption.REPLACE_EXISTING);
+		Files.deleteIfExists(Paths.get(filename));
+		Files.createFile(Paths.get(filename));
+
+		writeMappingsToFile(releaseNumber, stableIdsToOldIdsMappings, currentStableIdentifiers);
 
 		logger.info("MapOldStableIds finished");
 	}
 
-	// Iterate through returned results of DB IDs and stable IDs
-	private static Map<String, List<String>> getDbIdToStableIds(ResultSet resultSet) throws SQLException {
+	/**
+	 * Creates a MySQL driver that queries the stable_identifiers database for identifier and instanceId from the StableIdentifier table.
+	 * @param dba MySQLAdaptor, Not used to query the database, but to supply database parameters like host, username and password.
+	 * @return ResultSet from MySQL stable_identifiers database that consists of identifiers and their associated instanceId
+	 * @throws ClassNotFoundException - Thrown if MySQL driver class isn't found
+	 * @throws SQLException - Thrown if there are issues connecting/querying/interacting with stable_identifiers database
+	 */
+	private static ResultSet retrieveAllStableIdentifiers(MySQLAdaptor dba) throws ClassNotFoundException, SQLException {
+		// Need to use mysql driver to access stable_identifiers db
+		logger.info("Connecting to stable_identifiers db...");
+		Class.forName("com.mysql.jdbc.Driver");
+		Connection connect = DriverManager.getConnection("jdbc:mysql://" + dba.getDBHost() + "/stable_identifiers?" + "user=" + dba.getDBUser() + "&password=" + dba.getDBPwd());
+		Statement statement = connect.createStatement();
+		return statement.executeQuery("SELECT identifier,instanceId FROM StableIdentifier");
```

Comment:
Curious: Does this raw SQL query produce different results than using the Reactome API to get everything which is a `StableIdentifier`?

Reply:
It's actually not possible to use the Reactome API with the `stable_identifiers` database due to its schema

Reply:
Ah, ok.